### PR TITLE
test: add integration tests for DictionaryService

### PR DESCRIPTION
**Pull request by Abhineshhh**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [x] The PR title follows conventional commit label standards.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Summary

Adds comprehensive integration test coverage for `DictionaryService` with 9 test
cases covering all public methods (read, write, and validation operations).

## Related Issue
 #2576 

### Coverage Added
- insert() with valid data
- insert() with duplicate (exception handling)
- duplicateDictionaryExists() - true/false cases
- get() with invalid ID (edge case)
- Plus improved tests for existing methods with proper naming

### CODE REVIEW

File: src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
+package org.openelisglobal.dictionary.service;
```
Comment (Agaba-derrick) [inline src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceIntegrationTest.java]: @Abhineshhh this test  to some extent is a duplicate of [ this ](src/test/java/org/openelisglobal/dictionary/service) which i dont think you have taken a keen look at, 

my recomendation would be to look into the existing tests and the classes we are testing, look for which parts lack  coverage and test that, ensure before you create an issue especially on tests that it has not been worked upon before !

**Review by Agaba-derrick — CHANGES_REQUESTED** [review]

File: src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceIntegrationTest.java (hunk 1 above)
Comment (Abhineshhh) [reply src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceIntegrationTest.java]: Hi @Agaba-derrick, thanks for the review! 

I did check the existing `DictionaryServiceTest.java`  first
 ####  Here's what I found:
Missing in existing test (what my PR adds):
- `insert()` - completely untested
- `duplicateDictionaryExists()` - completely untested
- `get()` - invalid ID edge case
- Duplicate prevention exception handling

Yes there is overlap of some methods and i did it for the minimal overlap of methods and thought of consolidating later..

So What would you prefer now?

- Keep both
- Merge into one comprehensive file

Comment (Agaba-derrick) [reply src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceIntegrationTest.java]: Like i said , we check for missing tests then add them into existing testing classes , ideally we wouldnt want to have 2 files of  tests that test one class

Comment (Abhineshhh) [reply src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceIntegrationTest.java]: done 👍

File: src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java [hunk 2]
```diff
@@ -229,6 +229,61 @@ public void getPagesOfSearchedDictionaries_shouldFindByDictionaryText() {
...
+        dictionary.setIsActive("Y");
+        dictionary.setSortOrder(10);
+        dictionary.setSysUserId("admin");
+
+        try {
+            dictionaryService.insert(dictionary);
+            Assert.fail("Expected exception was not thrown for duplicate dictionary entry");
+        } catch (Exception e) {
+            // Expected behavior - duplicate entry should throw exception
+            Assert.assertTrue("Exception should be thrown for duplicate entry", true);
```
Comment (vsvishalsharma) [inline src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java]: These assertions always pass and don't verify anything. They should be removed since the test already validates behavior by reaching the catch block

File: src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java [hunk 3]
```diff
@@ -229,6 +229,61 @@ public void getPagesOfSearchedDictionaries_shouldFindByDictionaryText() {
...
+
+    @Test
+    public void testGetById_WithInvalidId_ReturnsNull() {
+        // Test that get() with invalid ID throws ObjectNotFoundException (expected behavior)
+        try {
+            dictionaryService.get("99999"); // Non-existent ID
+            Assert.fail("Expected ObjectNotFoundException for invalid ID");
+        } catch (org.hibernate.ObjectNotFoundException e) {
+            // Expected: Hibernate throws ObjectNotFoundException for invalid ID
+            Assert.assertTrue("Should throw ObjectNotFoundException for invalid dictionary ID", true);
```
Comment (vsvishalsharma) [inline src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java]: Catch the specific exception type (e.g., LIMSRuntimeException or ConstraintViolationException

**Review by vsvishalsharma — COMMENTED** [review]
if possible try addign some negative tests as well

File: src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java (hunk 2 above)
Comment (Abhineshhh) [reply src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java]: removed useless assertions

File: src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java (hunk 3 above)
Comment (Abhineshhh) [reply src/test/java/org/openelisglobal/dictionary/service/DictionaryServiceTest.java]: Changed line 244 from generic Exception to specific LIMSRuntimeException
Line 281 was already catching specific ObjectNotFoundException (not generic), just removed the useless assertion

**Comment by Abhineshhh** [comment]
> if possible try addign some negative tests as well

hey @vsvishalsharma ! negative tests are already addressed in : 
`testInsert_WithDuplicateEntry_ThrowsException`- tests invalid duplicate insertion
`testDuplicateDictionaryExists_WithUniqueEntry_ReturnsFalse` - tests non-existent entry
`testGetById_WithInvalidId_ThrowsObjectNotFoundException` - tests invalid ID lookup

can u be more specific about your needs about adressing negative tests ?

**Comment by Abhineshhh** [comment]
have a review @mozzy11

**Comment by Agaba-derrick** [comment]
@Abhineshhh can you update this branch  !

**Comment by Abhineshhh** [comment]
done @Agaba-derrick

**Comment by Agaba-derrick** [comment]
if you clean  out the unnecessary   comments this will be good !

**Comment by Agaba-derrick** [comment]
are you able to fix the CI failures !

**Review by mozzy11 — COMMENTED** [review]
Nice — you've kept this rebased against the current DictionaryServiceTest, and the additions are genuinely complementary: the duplicate-detection paths (insert throwing LIMSDuplicateRecordException, both duplicateDictionaryExists branches) and the nonexistent-id get() case aren't covered by the tests already on develop. We'll approve the CI run for the full pipeline; assuming it's green this looks mergeable as-is. One small ask: confirm the ObjectNotFoundException expectation still holds under the current Hibernate version — if CI passes, you're already there.
